tests/foreman/api/test_virt_who_config.py

```diff
@@ -0,0 +1,408 @@
+
+        :id: 4164bdde-f532-480c-b41e-747e87cf7d05
+        :steps:
+            1. Create a user with ONLY the virt-who manager role.
+            2. Verify the user can create virt-who configurations
+            3. Verify the user can edit an existing virt-who configuration
+            4. Verify the user can delete a virt-who configuration
+            5. Verify the user can see reporting info via the dashboard widget.
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_role_manager(self):
+        """virt-who Manager disallowed functions
+
+        :id: e93b415a-7442-4f27-9c52-15c72f3e1414
+
+        :steps:
+            1. Create a user with ONLY the virt-who manager role.
+            2. Verify the user can do no other actions other then those
+               in test_positive_role_manager
+
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_role_reporter(self):
+        """ Verify virt-who Reporter role
+
+        :id: 63462338-a6f5-48eb-8b04-433c53882817
+
+        :steps:
+            1. Create a user with ONLY the virt-who reporter role.
+            2. Configure virt-who with the user, WITHOUT using the
+               virt-who config plugin.
+            3. Create a vm to cause virt-who to send a report to satellite.
+            4. Verify the virt-who server send a report to the satellite.
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_role_reporter(self):
+        """Verify virt-who Reporter role
+
+        :id: 7dee0965-9ec4-4d76-a6ae-f2eec1960bac
+
+        :steps:
+            1. Verify the user can do no other actions other then those
+               in test_positive_role_reporter
+
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_role_viewer(self):
+        """ Verify virt-who viewer role
+
+        :id: 43314ae3-6768-44b5-a6bb-ee64ae381cd0
+
+        :steps:
+            1. Create a user with ONLY the virt-who Viewer role.
+            2. Verify the user can view virt-who configurations.
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_role_viewer(self):
+        """ Verify virt-who viewer role
+
+        :id: 6919a748-423e-4843-bafe-eb98b7159c90
+
+        :steps:
+            1. Create a user with ONLY the virt-who Viewer role.
+            2. Verify the user CANNOT delete or modify virt-who configurations
+            3. Verify the user can do no other actions other then those
+               in test_positive_role_viewer
+
+        """
+        pass
+
+
+class VirtWhoConfigUpgrade(APITestCase):
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_satellite_upgrade(self):
+        """ Satellite upgrade
+
+        :id: 0e301c08-8bef-4bea-a690-d4b0760949e8
+
+        :steps:
+            1. Start with a satellite version before the virt-who config plugin
```

Comment:
Didn't get this step..  do you mean "Install satellite w/o virt-who plugin ?"

Reply:
yes